subsys/debug/coredump/coredump_shell.c

```diff
@@ -0,0 +1,541 @@
+		shell_print(sh, "Unsupported command from the backend");
+	} else {
+		shell_print(sh, "Failed to perform verify command: %d", ret);
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Flush the print buffer to shell.
+ *
+ * This prints what is in the print buffer to the shell.
+ *
+ * @param sh shell instance.
+ */
+static void flush_print_buf(const struct shell *sh)
+{
+	shell_print(sh, "%s%s", COREDUMP_PREFIX_STR, print_buf);
+	print_buf_ptr = 0;
+	(void)memset(print_buf, 0, sizeof(print_buf));
+}
+
+/**
+ * @brief Helper to print Zephyr coredump header to shell
+ *
+ * @param sh shell instance
+ * @param buf pointer to header
+ * @return size of data processed, -EINVAL if error converting data
+ */
+static int print_coredump_hdr(const struct shell *sh, uint8_t *buf)
+{
+	struct coredump_hdr_t *hdr = (struct coredump_hdr_t *)buf;
+
+	if (memcmp(hdr->id, "ZE", sizeof(char)*2) != 0) {
+		shell_print(sh, "Not a Zephyr coredump header");
+		return -EINVAL;
+	}
+
+	shell_print(sh, "**** Zephyr Coredump ****");
+	shell_print(sh, "\tVersion %u", hdr->hdr_version);
+	shell_print(sh, "\tTarget: %s",
+		    coredump_target_code2str[sys_le16_to_cpu(
+				    hdr->tgt_code)]);
+	shell_print(sh, "\tPointer size: %u",
+		    (uint16_t)BIT(hdr->ptr_size_bits));
+	shell_print(sh, "\tFlag: %u", hdr->flag);
+	shell_print(sh, "\tReason: %u", sys_le16_to_cpu(hdr->reason));
+
+	return sizeof(struct coredump_hdr_t);
+}
+
+/**
+ * @brief Helper to print stored coredump data to shell
+ *
+ * This converts the binary data in @p buf to hexadecimal digits
+ * which can be printed to the shell.
+ *
+ * @param sh shell instance
+ * @param buf binary data buffer
+ * @param len number of bytes in buffer to be printed
+ * @return 0 if no issues; -EINVAL if error converting data
+ */
+static int print_stored_dump(const struct shell *sh, uint8_t *buf, size_t len)
+{
+	int ret = 0;
+	size_t i = 0;
+	size_t remaining = len;
+
+	if (len == 0) {
+		/* Flush print buffer */
+		flush_print_buf(sh);
+		goto out;
+	}
+
+	while (remaining > 0) {
+		if (hex2char(buf[i] >> 4, &print_buf[print_buf_ptr]) < 0) {
+			ret = -EINVAL;
+			break;
+		}
+		print_buf_ptr++;
+
+		if (hex2char(buf[i] & 0xf, &print_buf[print_buf_ptr]) < 0) {
+			ret = -EINVAL;
+			break;
+		}
+		print_buf_ptr++;
+
+		remaining--;
+		i++;
+
+		if (print_buf_ptr == PRINT_BUF_SZ) {
+			flush_print_buf(sh);
+		}
+	}
+
+out:
+	return ret;
+}
+
+/**
+ * @brief Helper parsing and pretty-printing the coredump
+ *
+ * @param sh shell instance
+ * @param coredump_header Pointer to a boolean indicating zephyr coredump
+ *        got parsed/printed aleardy header
+ * @param copy A pointer on the coredump copy context
+ * @param left_size How much of the coredump has not been parsed/printed yet
+ * @return 0 if all has been processed, a positive value indicating the amount
+ *        that the function call has processed, a negative errno otherwise.
+ */
+static int parse_and_print_coredump(const struct shell *sh,
```

Comment:
I believe that core dumps printed out in this new format will not work as expected with the existing tooling. The parser is expecting the dump in a certain format (hex print of full dump including the various headers).

https://github.com/zephyrproject-rtos/zephyr/blob/459a078d22607a0007ff3f4a2ef31c29c71a162f/scripts/coredump/coredump_serial_log_parser.py

and

https://github.com/zephyrproject-rtos/zephyr/blob/459a078d22607a0007ff3f4a2ef31c29c71a162f/scripts/coredump/coredump_gdbserver.py

Reply:
Didn't spot these tools. There are no tests for these, it seems.

I will then add an option "pretty" to print command, so default will be the previous behavior.